apps/samplePoint/sample-point.cpp

```diff
@@ -0,0 +1,126 @@
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "../commandLineApplication.h"
+
+#include <fivox/fivox.h>
+#include <fstream>
+
+class SamplePoint : public CommandLineApplication
+{
+public:
+    SamplePoint()
+        : _outputFile( "point_values.txt" )
+    {
+        _options.add_options()
+//! [SamplePointParameters] @anchor SamplePoint
+            ( "point,p", po::value< fivox::Vector3f >(),
+              "Point used as reference to extract the time series after "
+              "evaluating the event source for all specified frames")
+            ( "output,o", po::value< std::string >(),
+              "Name of the output file, containing one line per value, in the "
+              "format \"timestamp value\". Also a header with information "
+              "about the generation: volume URI, dt, frame range and the point "
+              "that was sampled" );
+//! [SamplePointParameters]
+    }
+
+    bool parse( int argc, char* argv[] ) final
+    {
+        if( !CommandLineApplication::parse( argc, argv ))
+            return false;
+
+        if( _vm.count( "output" ))
+            _outputFile = _vm["output"].as< std::string >();
+
+        if( _vm.count( "point" ))
+            _point = _vm["point"].as< fivox::Vector3f >();
+
+        return true;
+    }
+
+    bool write() const
+    {
+        const ::fivox::URIHandler params( getURI( ));
+        if( params.getDt() > -1.f )
+        {
+            LBWARN << "Specifying a custom dt in the volume URI is not "
+                   << "supported by the sample-point tool" << std::endl;
+            return EXIT_FAILURE;
+        }
+
+        ImageSourcePtr source = params.newImageSource< float >();
+        ImageSource::FunctorPtr functor = source->getFunctor();
+
+        const float dt = functor->getSource()->getDt();
+        const fivox::Vector2ui frameRange( getFrameRange( dt ));
+
+        typename fivox::FloatVolume::SpacingType spacing;
+        typename fivox::FloatVolume::PointType itkPoint;
+        itkPoint[0] = _point[0];
+        itkPoint[1] = _point[1];
+        itkPoint[2] = _point[2];
+
+        std::ofstream file( _outputFile );
+        file << "# File generated by the sample-point tool:\n"
+             << "# - Format: timestamp value\n"
+             << "# - Fivox URI: " << getURI() << "\n"
+             << "# - dt: " << dt << "\n"
+             << "# - Frame range: " << frameRange << "\n"
+             << "# - Point sampled: " << _point << "\n"
+             << std::endl;
+
+        for( uint32_t i = frameRange.x(); i < frameRange.y(); ++i )
+        {
+            functor->getSource()->load( i );
+            const float value = (*functor)( itkPoint, spacing );
```

Comment:
Pass a default constructed fivox::FloatVolume::SpacingType and remove the spacing variable?